### DISCUSSION TITLE
How to show the names of related existing items?

### DISCUSSION THREAD

**Discussion by willemmulder** [formfcw/directus-extension-flexible-editor#35] [Q&A] [answered]
So from what I understand:

We have a Pages Editor Nodes collection, that can relate any item to a specific M2A Editor field. On the M2A field, we can specify which items to support. 

- If we select a 'normal object' like Pages, then in the editor UI, we can only *create* a new Page and relate it
- If we want to select existing content, we'd use a new collection 'Related pages' with an M2M field that would point to different pages. Effectively, in the editor we would create a new record on that Related pages collection, and then select our existing pages. from there 

Now my question is: how can I ensure that in the editor, the names of those existing pages are displayed? 

I tried setting the Display Template of the "Related pages" object to Pages > Pages ID > Title, but that doesn't work. It shows -. Any clue?

**Comment by formfcw** [comment, marked as answer]
Thanks, this is a really good question. Even though it is a Directus related question, I will try my best to answer it here.

If you were just relating to a single page, it would be easy because you would use an M2O field instead and just use the title of the M2O item.

It gets more complex with a M2M or M2A field.

The fact, that `Pages > Pages ID > Title` shows a `--` seems to be a bug in Directus. But either way it would not be the solution because it would show a `X items` button, which would open a popover with the titles.

If you want to display the titles directly, you need to create a custom display extension or an action-hook or a flow (on create + update) that computes the titles into a new (hidden) field (inside the `related_pages` collection), so that you can display the contents of this extra field instead. This would mean some extra work, but I don’t see any other way in Directus at the moment.

**Reply by willemmulder** [reply]
Thanks a lot for the extensive answer @formfcw ! 

I'll also try to open an issue with Directus, so they can address this issue directly. Let's see what they come up with!
In the meantime, either we live with the - or I'll go with the flow-solution and calculate the field whenever necessary. 

Thanks again!
